rllib/models/experimental/torch/primitives.py

```diff
@@ -74,24 +73,21 @@ def __init__(
         self.input_dim = input_dim
         hidden_layer_dims = hidden_layer_dims
 
-        activation_class = getattr(nn, hidden_layer_activation, lambda: None)()
+        activation = getattr(nn, hidden_layer_activation, lambda: None)()
+
         layers = []
-        layers.append(nn.Linear(input_dim, hidden_layer_dims[0]))
-        for i in range(len(hidden_layer_dims) - 1):
+        dims = [input_dim] + hidden_layer_dims + [output_dim]
+        layers.append(nn.Linear(dims[0], dims[1]))
+        for i in range(1, len(dims) - 1):
             if hidden_layer_activation != "linear":
-                layers.append(activation_class)
-            layers.append(nn.Linear(hidden_layer_dims[i], hidden_layer_dims[i + 1]))
+                layers.append(activation)
+            layers.append(nn.Linear(dims[i], dims[i + 1]))
 
-        if output_dim is not None:
-            if hidden_layer_activation != "linear":
-                layers.append(activation_class)
-            layers.append(nn.Linear(hidden_layer_dims[-1], output_dim))
-            self.output_dim = output_dim
-        else:
-            self.output_dim = hidden_layer_dims[-1]
+        self.output_dim = dims[-1]
 
         if output_activation != "linear":
-            layers.append(get_activation_fn(output_activation, framework="torch"))
+            activation = getattr(nn, output_activation, lambda: None)()
+            layers.append(activation)
 
         self.mlp = nn.Sequential(*layers)
```

Comment:
This file is not waterproof and I'll construct tests that sweep over model configs very soon. For now, let's please go with few stupid configs for the RLTrainer test and other integration tests until we have settled on the majority of the Model and Catalog API.